00_basics/sc_dimensions.ipynb

```diff
@@ -0,0 +1,771 @@
+   "source": [
+    "Lets see what happens when we multiply non-matching dimensions of our arrays."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "humanitarian-robin",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-> [ 8, 15, 12 ]"
+     ]
+    }
+   ],
+   "source": [
+    "[2,3]*[4,5,6]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "considerable-might",
+   "metadata": {},
+   "source": [
+    "A bit unexpecting that we do not receive an error - instead SuperCollider tries to fix our error here by cycling the smaller array until we have gone through all arrays once fully.\n",
+    "Lets see how this cascades when we want to multiply 3 arrays with different size."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "analyzed-violin",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-> [ 56, 120, 108, 80 ]"
+     ]
+    }
+   ],
+   "source": [
+    "[2, 3] * [4, 5, 6] * [7, 8, 9, 10]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "addressed-warrant",
+   "metadata": {},
+   "source": [
+    "The results are calculated as following:\n",
+    "\n",
+    "$$\n",
+    "\\begin{aligned}\n",
+    "2 \\cdot 4 \\cdot 7 &= 7\\\\\n",
+    "3 \\cdot 5 \\cdot 8 &= 120\\\\\n",
+    "2 \\cdot 6 \\cdot 9 &= 108\\\\\n",
+    "3 \\cdot 4 \\cdot 10 &= 80\n",
+    "\\end{aligned}\n",
+    "$$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "entertaining-appliance",
+   "metadata": {},
+   "source": [
+    "### Matrix\n",
+    "\n",
+    "Now we can also take a look at how a matrix is represented in SuperCollider."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "biological-paper",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-> [ [ 2, 3, 4 ], [ 4, 5, 6 ] ]"
+     ]
+    }
+   ],
+   "source": [
+    "n = [\n",
+    "    [2, 3, 4],\n",
+    "    [4, 5, 6],\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "sorted-testing",
+   "metadata": {},
+   "source": [
+    "So it is basically just an array of arrays - but lets take a closer look at this as it is important to get the indices and the dimensions properly when working with matrices.\n",
+    "\n",
+    "Lets say we want to represent a $3 \\times 2$ matrix \n",
+    "$$\n",
+    "\\text{M} =\n",
+    "\\begin{pmatrix}\n",
+    "   1 & 2\\\\\n",
+    "   3 & 4\\\\\n",
+    "   5 & 6\n",
+    "\\end{pmatrix}\n",
+    "$$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "metric-glance",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[ [ 1, 2 ], [ 3, 4 ], [ 5, 6 ] ]\n",
+      "-> 2"
+     ]
+    }
+   ],
+   "source": [
+    "m = [\n",
```

Comment:
I think this is the wrong way round. The inner arrays are the columns!

Reply:
I think its the other way around
![image](https://user-images.githubusercontent.com/8267062/115020530-01ba6d80-9ebb-11eb-88d1-c56730c96469.png)
This way we can use the notation of first the index of the row, then of the column which is how it is the convention in mathematics as well